rules/private/phases/phase_javainfo.bzl

```diff
@@ -29,7 +29,7 @@ def phase_javainfo(ctx, g):
     scala_configuration_runtime_deps = _collect(JavaInfo, g.init.scala_configuration.runtime_classpath)
 
     if len(ctx.attr.srcs) == 0 and len(ctx.attr.resources) == 0:
-        java_info = java_common.merge([g.classpaths.sdeps, sexports])
+        java_info = java_common.merge([g.classpaths.sdeps, sexports, sruntime_deps])
```

Comment:
Does it matter that these are not passed to the different fields of the `JavaInfo`?

Reply:
I'm not sure what you mean. In the `else`? They're included in the `runtime_deps` on line 66. I don't have a complete understanding of what's being done here, but this change fixed the bug I was experiencing, where a `scala_binary` with an empty `srcs` list would ignore `runtime_deps` but include `deps`

Reply:
Yeah, in the else we do 
```
JavaInfo(
    exports = [sexports],
    runtime_deps = [sruntime_deps],
    deps = [g.classpaths.sdeps],
)
```
so I wasn't sure if that's significant or important. Maybe `java_common.merge` is doing that for us.